src/ddsrt/include/dds/ddsrt/event.h

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright(c) 2006 to 2020 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#ifndef EVENT_H
+#define EVENT_H
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+/*currently supported monitorable quantities*/
+enum MONITORABLE_TYPE {
+    MONITORABLE_UNSET,
+    MONITORABLE_FILE,
+    MONITORABLE_SOCKET,
+    MONITORABLE_PIPE,
+    MONITORABLE_VOLATILE
+};
+
+/*currently supported types of events*/
+enum EVENT_TYPE {
+    EVENT_UNSET = 0x00000000,
+    EVENT_DATA_IN = 0x00000001,
+    EVENT_DATA_OUT = 0x00000002,
+    EVENT_CONNECT = 0x0000004,
+    EVENT_DISCONNECT = 0x00000008,
+    EVENT_TIMEOUT = 0x00000010,
+    EVENT_ERROR = 0x80000000
+};
+
+/**
+* ddsrt_event struct definition
+* 
+* mon_type: type of monitorable for events
+* mon_ptr: pointer to stored value of monitorable
+* mon_sz: number of bytes stored by mon_ptr
+* evt_type: type of event, composited from EVENT_TYPE
+*/
```

Comment:
Don't know if `doxygen` will accept this (don't think so, but haven't tested), but please use the documentation style that's used in the rest of the files.